source/standalone/workflows/ray/isaac_ray_util.py

```diff
@@ -0,0 +1,409 @@
+    cfg: dict,
+    python_cmd: str = "/workspace/isaaclab/isaaclab.sh -p",
+    workflow: str = "source/standalone/workflows/rl_games/train.py",
+) -> str:
+    """Generate command with proper Hydra arguments"""
+    runner_args = []
+    hydra_args = []
+
+    def process_args(args, target_list, is_hydra=False):
+        for key, value in args.items():
+            if not is_hydra:
+                if key.endswith("_singleton"):
+                    target_list.append(value)
+                elif key.startswith("--"):
+                    target_list.append(f"{key} {value}")  # Space instead of = for runner args
+                else:
+                    target_list.append(f"{value}")
+            else:
+                if isinstance(value, list):
+                    # Check the type of the first item to determine formatting
+                    if value and isinstance(value[0], dict):
+                        # Handle list of dictionaries (e.g., CNN convs)
+                        formatted_items = [f"{{{','.join(f'{k}:{v}' for k, v in item.items())}}}" for item in value]
+                    else:
+                        # Handle list of primitives (e.g., MLP units)
+                        formatted_items = [str(x) for x in value]
+                    target_list.append(f"'{key}=[{','.join(formatted_items)}]'")
+                elif isinstance(value, str) and ("{" in value or "}" in value):
+                    target_list.append(f"'{key}={value}'")
+                else:
+                    target_list.append(f"{key}={value}")
+
+    print(f"[INFO]: Starting workflow {workflow}")
+    process_args(cfg["runner_args"], runner_args)
+    print(f"[INFO]: Retrieved workflow runner args: {runner_args}")
+    process_args(cfg["hydra_args"], hydra_args, is_hydra=True)
+    print(f"[INFO]: Retrieved hydra args: {hydra_args}")
+
+    invoke_cmd = f"{python_cmd} {workflow} "
+    invoke_cmd += " ".join(runner_args) + " " + " ".join(hydra_args)
+    return invoke_cmd
+
+
+@ray.remote
+def remote_execute_job(
+    job_cmd: str, identifier_string: str, test_mode: bool = False, extract_experiment: bool = False
+) -> str | dict:
+    """This method has an identical signature to :meth:`execute_job`, with the ray remote decorator"""
+    return execute_job(
+        job_cmd=job_cmd, identifier_string=identifier_string, test_mode=test_mode, extract_experiment=extract_experiment
+    )
+
+
+def execute_job(
+    job_cmd: str,
+    identifier_string: str = "job 0",
+    test_mode: bool = False,
+    extract_experiment: bool = False,
+    persistent_dir: str | None = None,
+    log_all_output: bool = False,
+) -> str | dict:
+    """Issue a job (shell command).
+
+    Args:
+        job_cmd: The shell command to run.
+        identifier_string: What prefix to add to make logs easier to differentiate
+            across clusters or jobs. Defaults to "job 0".
+        test_mode: When true, only run 'nvidia-smi'. Defaults to False.
```

Comment:
When would a user run test_mode?

Reply:
add comment. It's basically to sanity check